File: android/src/main/java/com/oblador/keychain/KeychainPackage.java
package com.oblador.keychain;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.JavaScriptModule;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.oblador.keychain.workaround.Issue314Filter;
import com.oblador.keychain.workaround.Issue337Filter;

import java.util.Collections;
import java.util.List;

@SuppressWarnings("unused")
public class KeychainPackage implements ReactPackage {

  public KeychainPackage() {

  }

  @Override
  @NonNull
  public List<NativeModule> createNativeModules(@NonNull final ReactApplicationContext reactContext) {
    return Collections.singletonList(new KeychainModuleBuilder(reactContext)
        .workaroundAffectedDevices(
          new Issue314Filter(),
          new Issue337Filter())
        .build());
  }

  @NonNull
  public List<Class<? extends JavaScriptModule>> createJSModules() {
    return Collections.emptyList();
  }

  @Override
  @NonNull
  public List<ViewManager> createViewManagers(@NonNull final ReactApplicationContext reactContext) {
    return Collections.emptyList();
  }
}

File: android/src/main/java/com/oblador/keychain/KeychainPackage.kt
package com.oblador.keychain

import com.facebook.react.TurboReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider

@Suppress("unused")
class KeychainPackage : TurboReactPackage() {

  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == KeychainModule.KEYCHAIN_MODULE) {
      KeychainModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      val moduleInfos: MutableMap<String, ReactModuleInfo> = HashMap()
      val isTurboModule: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
      moduleInfos[KeychainModule.KEYCHAIN_MODULE] = ReactModuleInfo(
        KeychainModule.KEYCHAIN_MODULE,
        KeychainModule::class.java.name,
        false, // canOverrideExistingModule
        false, // needsEagerInit
        false, // isCxxModule
        isTurboModule // isTurboModule
      )
      moduleInfos
    }
  }
}
